src/main/java/no/nav/familie/integrasjoner/dokarkiv/DokarkivService.kt

```diff
@@ -41,14 +46,28 @@ class DokarkivService(private val dokarkivRestClient: DokarkivRestClient,
 
         val metadata = dokarkivMetadata.getMetadata(arkiverDokumentRequest.hoveddokumentvarianter[0])
         val hoveddokument = mapHoveddokument(arkiverDokumentRequest.hoveddokumentvarianter)
-        val vedleggsdokumenter = arkiverDokumentRequest.vedleggsdokumenter.map(this::mapTilArkivdokument)
+        var vedleggsdokumenter = arkiverDokumentRequest.vedleggsdokumenter.map(this::mapTilArkivdokument)
         val jpsak: Sak? = if (arkiverDokumentRequest.fagsakId != null)
             Sak(fagsakId = arkiverDokumentRequest.fagsakId,
                 sakstype = "FAGSAK",
                 fagsaksystem = metadata.fagsakSystem) else null
 
         val navn = hentNavnForFnr(fnr = arkiverDokumentRequest.fnr, behandlingstema = metadata.tema)
 
+        //førsteside
+        val førsteside: ByteArray? = if (arkiverDokumentRequest.førsteside != null)
+            førstesideGeneratorService.genererForside(arkiverDokumentRequest.førsteside!!) else null
+
+        if(førsteside != null) {
+            vedleggsdokumenter += ArkivDokument(brevkode = metadata.brevkode,
+                    dokumentKategori = metadata.dokumentKategori,
+                    tittel = arkiverDokumentRequest.førsteside?.overskriftsTittel,
+                    dokumentvarianter = listOf(DokumentVariant(filtype = "Førsteside",
+                            variantformat = "pdf",
+                            fysiskDokument = førsteside,
+                            filnavn = "førsteside.pdf")))
+        }
+
```

Comment:
```suggestion

        if (arkiverDokumentRequest.førsteside != null) {
           val førsteside = førstesideGeneratorService.genererForside(arkiverDokumentRequest.førsteside
            vedleggsdokumenter += ArkivDokument(brevkode = metadata.brevkode,
                    dokumentKategori = metadata.dokumentKategori,
                    tittel = arkiverDokumentRequest.førsteside?.overskriftsTittel,
                    dokumentvarianter = listOf(DokumentVariant(filtype = "Førsteside",
                            variantformat = "pdf",
                            fysiskDokument = førsteside,
                            filnavn = "førsteside.pdf")))
        }

```

Kunne evt dette vært noe?

Reply:
👍  er ryddigere, men da risikerer vi at førsteside er null når vi skal legge til arkivdokument, så mulig det blir litt mindre robust kode?